.github/workflows/black.yaml

```diff
@@ -0,0 +1,29 @@
+name: Black Formatting
+
+on: [pull_request, push]  # NOLINT
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.6, 3.7, 3.8]
+
+    steps:
+    - uses: actions/checkout@v2  # NOLINT
+
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip3 install -r requirements.txt
+        pip3 install black
+
+    - name: Test with black
+      run: >-
+        black --check --diff src tests
```

Comment:
This could run into some problems for unit tests, where you have files that intentionally have syntax or formatting errors. I suggest switching to the following way to run this test:

```
- name: Test with black
  run: >-
    black --check --diff
    --exclude tests/.*/*valid_package/.*
    src tests
```

Reply:
Thanks for the tip.